### DISCUSSION TITLE
Is it possible to use a custom Identity Server to integrate with Ocelot?

### DISCUSSION THREAD

**Discussion by Ewerton** [ThreeMammals/Ocelot#1850] [Q&A]
I read the [authentication page](https://ocelot.readthedocs.io/en/latest/features/authentication.html) and understood that Ocelot is designed to integrate with identity servers using the default protocols for that (OAuth, OpenId).

I'm not able to buy or configure an enterprise grade identity server right now so I'm wondering if it is possible to implement something like basic authentication where I could call a function or an API which will responds if the user is authenticated or not. If so, can you provide some code samples on how to implement it?

**Comment by raman-m** [comment]
Hi @Ewerton !
Welcome to Ocelot world! :tiger2: 

You can do in Ocelot mostly everything which is possible with ASP.NET Core. &rarr;

**OAuth**
- [Authenticate with external providers](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/secure-net-microservices-web-applications/#authenticate-with-external-providers)
- [External OAuth authentication providers](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/social/other-logins?view=aspnetcore-8.0)

**OpenId** etc.
- [Community OSS authentication options for ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/community)
